Content.Server/DeltaV/StationEvents/Events/GlimmerMiteSpawnRule.cs

```diff
@@ -0,0 +1,58 @@
+using System.Linq;
+using Robust.Shared.Random;
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.NPC.Components;
+using Content.Server.Psionics.Glimmer;
+using Content.Server.StationEvents.Components;
+using Content.Shared.Psionics.Glimmer;
+using Content.Shared.Abilities.Psionics;
+
+namespace Content.Server.StationEvents.Events;
+
+internal sealed class GlimmerMiteRule : StationEventSystem<GlimmerMiteRuleComponent>
+{
+    [Dependency] private readonly IRobustRandom _robustRandom = default!;
+    [Dependency] private readonly GlimmerSystem _glimmerSystem = default!;
+
+    [DataField(required: true)]
+    public EntProtoId MobPrototype;
+
+    protected override void Started(EntityUid uid, GlimmerMiteRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
+    {
+        base.Started(uid, component, gameRule, args);
+
+        var glimmerSources = EntityQuery<GlimmerSourceComponent, TransformComponent>().ToList();
+        var normalSpawnLocations = EntityQuery<VentCritterSpawnLocationComponent, TransformComponent>().ToList();
+        var hiddenSpawnLocations = EntityQuery<MidRoundAntagSpawnLocationComponent, TransformComponent>().ToList();
+
+        var baseCount = Math.Max(1, EntityQuery<PsionicComponent, NpcFactionMemberComponent>().Count() / 10);
+        int multiplier = Math.Max(1, (int) _glimmerSystem.GetGlimmerTier() - 2);
+
+        var total = baseCount * multiplier;
+
+        int i = 0;
+        while (i < total)
+        {
+            if (glimmerSources.Count != 0 && _robustRandom.Prob(0.4f))
+            {
+                Spawn(MitePrototype, _robustRandom.Pick(glimmerSources).Item2.Coordinates);
```

Comment:
then use `component.MobPrototype` for all these

Reply:
Alright, I *think* I fixed all these correctly